cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java

```diff
@@ -83,44 +65,39 @@ public double getPriceForInstanceType(String region, String instanceType) {
         return priceListElement.getProduct().getAttributes().getVcpu();
     }
 
+    @Cacheable(cacheNames = "awsCostCache", key = "{ #region, #instanceType }")
     public int getMemoryForInstanceType(String region, String instanceType, ExtendedCloudCredential extendedCloudCredential) {
         PriceListElement priceListElement = getPriceList(region, instanceType);
         String memory = priceListElement.getProduct().getAttributes().getMemory();
         return Integer.parseInt(memory.replaceAll("\\D", ""));
     }
 
+    @Cacheable(cacheNames = "awsCostCache", key = "{ #region, #storageType, #volumeSize }")
     public double getStoragePricePerGBHour(String region, String storageType, int volumeSize) {
         if (volumeSize == 0 || storageType == null) {
             LOGGER.info("The provided volumeSize is 0 or the storageType is null, so returning 0.0 as storage price per GBHour.");
             return 0.0;
         }
-        Map<String, Double> pricingForRegion = storagePricingCache.getOrDefault(region, Map.of());
+        Map<String, Double> pricingForRegion = loadStoragePriceList().getOrDefault(region, Map.of());
         double priceInGBMonth = pricingForRegion.getOrDefault(storageType, 0.0);
         return priceInGBMonth / HOURS_IN_30_DAYS;
     }
 
     private PriceListElement getPriceList(String region, String instanceType) {
-        PriceListElement value = cache.getIfPresent(new PricingCacheKey(region, instanceType));
-        if (value == null) {
-            GetProductsResult productsResult = getProducts(region, instanceType);
-            String priceListString = productsResult.getPriceList().stream().findFirst()
-                    .orElseThrow(() ->
-                            new NotFoundException(String.format("Couldn't find the price list for the region %s and instance type %s!", region, instanceType)));
-            try {
-                PriceListElement priceListElement = JsonUtil.readValue(priceListString, PriceListElement.class);
-                cache.put(new PricingCacheKey(region, instanceType), priceListElement);
-                return priceListElement;
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        } else {
-            LOGGER.info("PriceList for region [{}] and instance type [{}] found in cache.", region, instanceType);
-            return value;
+        GetProductsResult productsResult = getProducts(region, instanceType);
+        String priceListString = productsResult.getPriceList().stream().findFirst()
+                .orElseThrow(() ->
+                        new NotFoundException(String.format("Couldn't find the price list for the region %s and instance type %s!", region, instanceType)));
+        try {
+            return JsonUtil.readValue(priceListString, PriceListElement.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
could you replace this with a meaningful excpetion?

Reply:
is it OK if I'll do it in another PR? this is not a new code, only if condition was removed

Reply:
nvm, i'll fix it here